frog/server/accountManagement.js

```diff
@@ -0,0 +1,50 @@
+import { Accounts } from 'meteor/accounts-base';
+import { Meteor } from 'meteor/meteor';
+
+/**
+ * Creates a new user account if userid is null or upgrades adds the new information to the given userid and makes the given account 'verified'.
+ *
+ * @optional: {string} userid - current user
+ * @param: {string}, email - email to be associated with the acount
+ * @param: {string}, password - password of the new account
+ * @param: {Object}, profile - user profile which consists of the DisplayName
+ */
+
+export const createAccount = (
+  userid: string,
+  email: string,
+  password: string,
+  profile: Object
+) => {
+  const user = Meteor.user();
+
+  if (userid && user?.isAnonymous) {
+    // checks for duplicate email
+    if (!Accounts.findUserByEmail(email)) {
```

Comment:
What do we do if we find a duplicate email. Should we notify the user?

Reply:
Yes we will display an error message on the client. I will move most of this logic to the client when I build the UI.

Reply:
However this is still a case that might happen at the server level (as you implemented a conditional check for this exact scenario). I would suggest outputting a console.err to indicate this is an unwanted state.